lib/license_finder/package_manager.rb

```diff
@@ -10,44 +10,51 @@ module LicenseFinder
   #
   # - implement #current_packages, to return a list of `Package`s this package manager is tracking
   # - implement #possible_package_paths, an array of `Pathname`s which are the possible locations which contain a configuration file/folder indicating the package manager is in use.
+  # - implement(Optional) #prepare_command, this is the package manager command that gets run when the --prepare flag is passed to license_finder.
   #
   class PackageManager
-    def self.package_managers
-      [GoDep, GoWorkspace, Go15VendorExperiment, Glide, Gvt, Govendor, Dep, Bundler, NPM, Pip,
-       Yarn, Bower, Maven, Gradle, CocoaPods, Rebar, Nuget, Carthage, Mix, Conan]
-    end
-
-    def self.current_packages(options)
-      active_package_managers(options).flat_map(&:current_packages_with_relations)
-    end
+    class << self
+      def package_managers
+        [GoDep, GoWorkspace, Go15VendorExperiment, Glide, Gvt, Govendor, Dep, Bundler, NPM, Pip,
+         Yarn, Bower, Maven, Gradle, CocoaPods, Rebar, Nuget, Carthage, Mix, Conan]
+      end
 
-    def self.active_package_managers(options={:project_path => Pathname.new('')})
-      active_pm_classes = package_managers.select { |pm_class| pm_class.new(options).active? }
-      active_pm_classes -= active_pm_classes.map(&:takes_priority_over)
-      active_pm_classes.map { |pm_class| pm_class.new(options) }
-    end
+      def active_packages(options)
+        active_package_managers(options).flat_map(&:current_packages_with_relations)
+      end
 
-    def self.takes_priority_over
-      nil
-    end
+      def active_package_managers(options={:project_path => Pathname.new('')})
+        active_pm_classes = package_managers.select { |pm_class| pm_class.new(options).active? }
+        active_pm_classes -= active_pm_classes.map(&:takes_priority_over)
+        active_pm_classes.map { |pm_class| pm_class.new(options) }
+      end
 
-    def self.installed?(logger=Core.default_logger)
-      if package_management_command.nil?
-        logger.installed self, "no command defined" # TODO comment me out
-        return true
+      def takes_priority_over
+        nil
       end
 
-      if command_exists?(package_management_command)
-        logger.installed self, true
-        return true
+      def installed?(logger=Core.default_logger)
+        if package_management_command.nil?
+          logger.installed self, "no command defined" # TODO comment me out
+          return true
+        end
+
+        if command_exists?(package_management_command)
+          logger.installed self, true
+          return true
+        end
+
+        logger.installed self, false
+        return false
       end
 
-      logger.installed self, false
-      return false
-    end
+      def package_management_command
+        nil
+      end
 
-    def self.package_management_command
-      nil
+      def prepare_command
```

Comment:
Might be nice to have comments indicating which methods of PackageManager must/should be included in subclasses